src/Symfony/Component/DependencyInjection/Compiler/DecoratorServicePass.php

```diff
@@ -57,6 +57,7 @@ public function process(ContainerBuilder $container)
                 $public = $decoratedDefinition->isPublic();
                 $decoratedDefinition->setPublic(false);
                 $decoratedDefinition->setTags(array());
+                $decoratedDefinition->setAutowiringTypes(array());
```

Comment:
I think we should first add the old value to the decorating service. What do you think?